dali/pipeline/operators/argument.h

```diff
@@ -70,35 +86,10 @@ class ValueInst : public Value {
 
 template <typename T>
 inline std::unique_ptr<Value> Value::construct(const T& val) {
-  return std::unique_ptr<Value>(new ValueInst<T>(val));
+  using S = argument_storage_t<T>;
```

Comment:
As you already use `Value` in the line bellow, maybe:
```suggestion
  using Storage = argument_storage_t<T>;
```
A little bit more verbosity wouldn't hurt here imo.